core/src/token_info/cached.rs

```diff
@@ -0,0 +1,159 @@
+use super::{TokenBaseInfo, TokenId, TokenInfoFetching};
+
+use anyhow::Result;
+use futures::future::{BoxFuture, FutureExt};
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
```

Comment:
Since this is async code we have to use an async mutex: https://docs.rs/async-std/1.6.2/async_std/sync/struct.RwLock.html .